### DISCUSSION TITLE
Built in way to access battery charge state / charging/discharging status.

### DISCUSSION THREAD

**Discussion by rootexpression** [Seeed-Studio/OSHW-XIAO-Series#25] [Open for Vote]
I, and it seems a whole lot of other people would like to see battery level monitoring hardware built in.  For any compact, battery powered project, knowing the charge state is important (as well as whether its currently charging or not, for say, disabling sleep while charging, etc), but again if its a compact thing adding extra discrete components to monitor the charge state, and take up more of the very few broken out pins isn't often a good way to go.
There are a lot of pins not being used on most of these mcu's on the smaller boards as is, so maybe adding a pad that can be cut if someone doesn't want to risk extra power draw from a voltage divider, if you go with a voltage divider and not another method, might sort concerns for people who care more about pinching every microamp in their project.

**Comment by Chunchun-tian** [comment]
yes You are right, XIAO RP2350, XIAO RA4M1, XIAO MG24 all have built-in IO measuring voltage, you can try it
